lib/suite.js

```diff
@@ -92,6 +92,19 @@ function Suite(title, parentContext, isRoot) {
  */
 inherits(Suite, EventEmitter);
 
+/**
+ * Resets the state initially or for a next run.
+ */
+Suite.prototype.reset = function() {
+  this.delayed = false;
+  this.suites.forEach(function(suite) {
+    suite.reset();
+  });
+  this.tests.forEach(function(test) {
+    test.reset();
+  });
```

Comment:
what of hooks?

Reply:
I'm not 100% sure I know what you mean with hooks, but I think it is `beforeEach` and friends.

I super duper don't want them to be cleaned. They should stay, just like the test fn's themselves. We're going to need them for the next run.

Reply:
yes, hooks are `beforeEach` etc.  I think I am confused about why suites and tests are getting `reset()` called, but not hooks.

Reply:
to be clear:  I'm still confused.

Reply:
I think I now understand your criticism. Hooks have state as well. And they inherit from `Runnable`. Didn't realize. Will add it when I have some time. Thanks for pointing it out!